### PR TITLE
Abort on endstop hit improvement: custom G-Code feature

### PR DESCRIPTION
**Pull request by ExtNeon**
### Description
A small improvement which adds new advanced configuration item similar to `EVENT_GCODE_SD_ABORT`, but when endstops are hit while printing from SD card. Only applies when the `SD_ABORT_ON_ENDSTOP_HIT` is activated. If not defined, behaviour of the feature stays the same.

Tested on Ender 3 4.2.2. 

### Requirements
None.

### Benefits
Ability to correctly stop print process when bad things happened. Current implementation of this feature only stops print and does nothing else, E.G. doesn't move the nozzle away, alert user, etc.

### Configurations
N/A

### Related Issues
None.

### CODE REVIEW

File: Marlin/Configuration_adv.h [hunk 1]
```diff
@@ -1555,6 +1555,11 @@
    */
   //#define SD_ABORT_ON_ENDSTOP_HIT
 
+  /**
+   * Run custom G-Code when endstop hit occured.
+   */
+  //#define SD_ABORT_ON_ENDSTOP_HIT_CUSTOM_GCODE "M300 S1000 P500"
```
Comment (thisiskeithb) [inline Marlin/Configuration_adv.h]: I'd recommend grouping this under `//#define SD_ABORT_ON_ENDSTOP_HIT` and then wrap it with an `#if ENABLED(SD_ABORT_ON_ENDSTOP_HIT)`:

```cpp

  /**
   * Abort SD printing when any endstop is triggered.
   * This feature is enabled with 'M540 S1' or from the LCD menu.
   * Endstops must be activated for this option to work.
   */
  //#define SD_ABORT_ON_ENDSTOP_HIT
  #if ENABLED(SD_ABORT_ON_ENDSTOP_HIT)
    //#define SD_ABORT_ON_ENDSTOP_HIT_CUSTOM_GCODE "G28XY"  // G-code to run on endstop hit (e.g., "G28XY" or "G27")
  #endif
```

Note: I've modified the default G-code to match `EVENT_GCODE_SD_ABORT`.

File: Marlin/Configuration_adv.h [hunk 2]
```diff
@@ -1554,7 +1554,10 @@
    * Endstops must be activated for this option to work.
    */
   //#define SD_ABORT_ON_ENDSTOP_HIT
-
+  #if ENABLED(SD_ABORT_ON_ENDSTOP_HIT)
+    //#define SD_ABORT_ON_ENDSTOP_HIT_CUSTOM_GCODE "M300 S1000 P500" //Run custom G-Code when endstop hit occured.
```
Comment (thisiskeithb) [inline Marlin/Configuration_adv.h]: I'd recommend changing this to `//#define SD_ABORT_ON_ENDSTOP_HIT_CUSTOM_GCODE "G28XY"  // G-code to run on endstop hit (e.g., "G28XY" or "G27")` so it’s more in line with `EVENT_GCODE_SD_ABORT`.

Comment (ExtNeon) [reply Marlin/Configuration_adv.h]: Sure thing, already done, thank you!

File: Marlin/Configuration_adv.h [hunk 3]
```diff
@@ -1554,7 +1554,10 @@
    * Endstops must be activated for this option to work.
    */
   //#define SD_ABORT_ON_ENDSTOP_HIT
-
+  #if ENABLED(SD_ABORT_ON_ENDSTOP_HIT)
+    //#define SD_ABORT_ON_ENDSTOP_HIT_CUSTOM_GCODE "G28XY" //G-code to run on SD Abort Print (e.g., "G28XY" or "G27")
```
Comment (thisiskeithb) [inline Marlin/Configuration_adv.h]: I forgot to change the comment in my last reply. Sorry about that:

`// G-code to run on endstop hit (e.g., "G28XY" or "G27")`

Comment (ExtNeon) [reply Marlin/Configuration_adv.h]: Oh, i didn't seen that either. Fixing...
I'm new on that collaboration thing 😅

**Pull request merged**